File: 0925-long-pressed-name/0925-long-pressed-name.java
class Solution {
    public boolean isLongPressedName(String name, String typed) {
        int a=0;
        int b=0;
        while(a<name.length() && b<typed.length()){
            if(name.charAt(a)==typed.charAt(b)){
                a++;
                b++;
            }
            else{
                if (b>0 && typed.charAt(b)==typed.charAt(b-1)){
                    b+=1;
                }
                else{
                    return false;
                }
            }
        }
        while(b<typed.length()){
            if(b>0 && typed.charAt(b)==typed.charAt(b-1)){
                b+=1;
            }
            else{
                return false;
            }
        }
        return a==name.length();

        
        
    }
}

File: 0925-long-pressed-name/0925-long-pressed-name.py
class Solution(object):
    def isLongPressedName(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        a,b=0,0
        while a<len(s) and b<len(t):
            if s[a]==t[b]:
                a+=1
                b+=1
            else:
                if b>0 and t[b]==t[b-1]:
                    b+=1
                else:
                    return False
        while b<len(t):
            if b>0 and t[b]==t[b-1]:
                b+=1
            else:
                return False            
        return a==len(s)
        